core/savings.go

```diff
@@ -0,0 +1,86 @@
+package core
+
+import (
+	"math"
+	"time"
+
+	"github.com/evcc-io/evcc/util"
+)
+
+// Site is the main configuration container. A site can host multiple loadpoints.
+type Savings struct {
+	log *util.Logger
+
+	startupTime            time.Time // Boot time
+	lastUpdateTime         time.Time // Time of last charged value update
+	chargedTotal           float64   // Energy charged since startup (Wh)
+	chargedSelfConsumption float64   // Self-produced energy charged since startup (Wh)
+}
+
+// NewSite creates a Site with sane defaults
+func NewSavings() *Savings {
+	savings := &Savings{
+		log:            util.NewLogger("savings"),
+		startupTime:    time.Now(),
+		lastUpdateTime: time.Now(),
+	}
+
+	return savings
+}
+
+func (s *Savings) Since() int {
```

Comment:
darum keine time.Duration?

Reply:
Good Point. Bin im Kopf noch ab und an in JS-Land :D